### ISSUE TITLE
scripting (lua,js... and etc)

### ISSUE THREAD

**Issue by seniorivn** [xylosper/bomi#246]
is it possible to make bomi sciptable like vlc?

**Comment by xylosper** [comment]
Well, it's quite technicall question.
First of all, why do you need it?
If you need a feature, you can just request it.

bomi utilizes QML for user interface, so you may think it's scriptable.
But, the API is mainly aimed for skin design, so what you can do is somewhat limited.

**Comment by seniorivn** [comment]
I was thinking about integration with anki, but has not yet decided how it should look like, ideally probably need to add all the words of the subtitle in Anki, verify the presence of base, etc.
Question about scripting appeared after viewing solutions for vlc.

**Comment by xylosper** [comment]
Because Qt can evaluate javascript(QML), if I support scripting functionality, it will be confined to javascript/QML only.
However, there are tons of things to be done. The most difficult thing is to define public APIs.
Even the skin API isn't fixed yet.
But, it's defintely interesting idea.

**Comment by xylosper** [comment]
When you saying 'scripting', what kind of functionality do you need?
Evaluating a script by bomi would be easy.
However, a script binding for bomi would be difficult, and I recommend writing a skin for such purpose.
A descriptive pseudo code would be most helpful.

**Comment by seniorivn** [comment]
My case is clickable words in subtitles and call dictionary and etc.

**Comment by xylosper** [comment]
Clickable words in subtitles 'by script'?
I cannot even imagine what you are talking about.
Do you want to make a 'click' by script?

**Comment by seniorivn** [comment]
no, i mean that when i will click word in subtitle, script will open this word in dictionary, or when i move mouse cursor to english language subtitle it will show native language subtitle, i just need interactive subtitles like here fore example(click to play video and then move cursor to subtitle and click the word) http://www.hamatata.com/

**Comment by xylosper** [comment]
This is not about scripting.
This is not possible fundamentally unless I replace current subtitlte renderer with totally differerent thing.
I'll keep this open but maybe it will take a long time till I start to work on this.

**Comment by seniorivn** [comment]
Ok, i'll solve it in external script, thanks.

**Comment by mac1202** [comment]
Mpv natively support script, you can find some example [here](https://github.com/mpv-player/mpv/wiki/User-Scripts), maybe you could make bomi able to use them.

**Comment by ahjolinna** [comment]
VapourSynth scripts would be fun to use, like: 
" [mvtools](https://github.com/haasn/gentoo-conf/blob/master/home/nand/.mpv/filters/mvtools.vpy) - Use [MVTools's](https://github.com/dubhater/vapoursynth-mvtools) BlockFPS function to perform motion interpolation on the video in realtime."
